### PR TITLE
Update the starter frontend install branch using release workflows

### PR DESCRIPTION
**Pull request by kennyadsl**
## Summary

This extra steps of the workflows will update the version on the starter frontend install template two times:

1. during the pre-release workflow, it will change it to the new candidate version to release. This way, when we will release we will have the correct version.
2. during the post-release workflow, it will change it back to main so that in development we will continue to target the dev branch of the starter frontend.

Closes https://github.com/solidusio/solidus/issues/4899

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: .github/workflows/prepare_release.yml [hunk 1]
```diff
@@ -32,6 +32,11 @@ jobs:
         run: |
           cd dev_tools/ && bin/bump-docker-image-version \
             ${{ steps.pipeline_context.outputs.candidate_version }}
+      - name: "Bump installer starter frontend version"
+        if: github.ref_name == 'master'
+        run: |
+          cd dev_tools/ && bin/bump-starter-frontend-version \
+            ${{ steps.pipeline_context.outputs.candidate_version }}
```
Comment (waiting-for-dev) [inline .github/workflows/prepare_release.yml:39]: I think we should use [`candidate_patch_branch`](https://github.com/solidusio/solidus/blob/04a4d68ef9c6957a88e6d66618397754e58367c0/dev_tools/lib/solidus/pipeline_context.rb#L72). `candidate_version` would return `3.4.0`, while `candidate_patch_branch` would return `v3.4`.

File: dev_tools/bin/bump-starter-frontend-version [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative '../lib/solidus/bumper'
+
+candidate_version = ARGV[0]
+
+path = File.expand_path('../../core/lib/generators/solidus/install/app_templates/frontend/starter.rb', __dir__)
+from = File.read(path)[/solidusio\/solidus_starter_frontend\/raw\/(.+)\/template.rb/, 1]
```
Comment (waiting-for-dev) [inline dev_tools/bin/bump-starter-frontend-version:9]: I'm failing to understand how it works :thinking:

File: .github/workflows/prepare_release.yml (hunk 1 above)
Comment (kennyadsl) [reply .github/workflows/prepare_release.yml:39]: You are rigth!

File: dev_tools/bin/bump-starter-frontend-version (hunk 2 above)
Comment (kennyadsl) [reply dev_tools/bin/bump-starter-frontend-version:9]: From the file content, it matches the regular expression and returns what matches the thing inside the parenthesis, in this case `(.+)`, which will be `main` or the branch name (eg. `v3.2`).

File: .github/workflows/prepare_release.yml (hunk 1 above)
Comment (kennyadsl) [reply .github/workflows/prepare_release.yml]: Fixed!

File: dev_tools/bin/bump-starter-frontend-version (hunk 2 above)
Comment (kennyadsl) [reply dev_tools/bin/bump-starter-frontend-version:9]: I did this test, in the sandbox:

```ruby
path = File.expand_path('../core/lib/generators/solidus/install/app_templates/frontend/starter.rb', __dir__)
File.read(path)[/solidusio\/solidus_starter_frontend\/raw\/(.+)\/template.rb/, 1]
# => "main"
```

Comment (waiting-for-dev) [reply dev_tools/bin/bump-starter-frontend-version:9]: Oh, sorry :facepalm: First review on Monday :sweat:

Comment (kennyadsl) [reply dev_tools/bin/bump-starter-frontend-version:9]: No need to say sorry! It's probably a signal that it's not super intuitive, and I'm open to more readable ways of expressing this if you have any suggestions.

**Review by rainerdema — APPROVED** [review]

File: dev_tools/bin/bump-starter-frontend-version (hunk 2 above)
Comment (kennyadsl) [reply dev_tools/bin/bump-starter-frontend-version]: Done!

**Pull request merged**